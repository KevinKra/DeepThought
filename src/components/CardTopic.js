import React from "react";
import "../styles/_CardTopic.scss";

class CardTopic extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      topic: this.props.topic
    };
  }
  handleClick = () => {
    // console.log("hello");
    // console.log(this.props.history);
    this.props.history.push(`/review/review-${this.state.topic}`);
  };
  render() {
    return (
      <section className="card-topic" onClick={this.handleClick}>
        <h1 className="card-title">{this.props.topic}</h1>
        <p className="card-body">{this.props.blurb}</p>
      </section>
    );
  }
}

export default CardTopic;
