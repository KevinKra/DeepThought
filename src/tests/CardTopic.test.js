import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CardTopic from "../components/CardTopic";

const mockTopic = "React";
const mockRenderReview = jest.fn();
const mockRenderNotFound = jest.fn();

const defaultState = {
  topic: mockTopic
};

describe("<CardTopic />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CardTopic topic="React" />);
  });
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should match default state", () => {
    expect(wrapper.state()).toEqual(defaultState);
  });
  // it("should invoke renderReview if the local state topic equals 'React'", () => {
  //   wrapper = shallow(<CardTopic topic="React" />);
  //   wrapper.find("section").simulate("click");
  //   expect(wrapper.instance().mockHandleClick).toEqual(mockRenderReview);
  // });
});
