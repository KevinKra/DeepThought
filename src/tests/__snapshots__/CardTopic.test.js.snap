// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTopic /> should match the snapshot 1`] = `
<section
  className="card-topic"
  onClick={[Function]}
>
  <h1
    className="card-title"
  >
    React
  </h1>
  <p
    className="card-body"
  />
</section>
`;
